plugins/dialog/plugin.js

```diff
@@ -3338,12 +3435,20 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 						if ( typeof dialogDefinition != 'function' )
 							CKEDITOR.dialog._.dialogDefinitions[ dialogName ] = 'failed';
 
-						this.openDialog( dialogName, callback );
+						this.openDialog( dialogName, callback, forceModel );
 					}, this, 0, 1 );
 			}
 
 			CKEDITOR.skin.loadPart( 'dialog' );
 
+			// Dissolve model, so `definition.getModel` can take precedence
+			// in the next dialog opening (#2423).
+			if ( dialog ) {
+				dialog.once( 'hide', function() {
```

Comment:
Probably it would be safer to add priority of `999` to be sure to not mess with some custom user logic.